dessia_common/core.py

```diff
@@ -491,12 +491,37 @@ def mpl_plot(self, **kwargs):
             msg = f"Class '{self.__class__.__name__}' does not implement a plot_data method to define what to plot"
             raise NotImplementedError(msg)
         return axs
-
-    @staticmethod
-    def display_settings() -> List[DisplaySetting]:
+    
+    @classmethod
+    def display_settings(cls) -> List[DisplaySetting]:
         """ Return a list of objects describing how to call object displays. """
-        return [DisplaySetting(selector="markdown", type_="markdown", method="to_markdown", load_by_default=True),
-                DisplaySetting(selector="plot_data", type_="plot_data", method="plot_data", serialize_data=True)]
+        list_display_settings = [DisplaySetting(selector="markdown", type_="markdown", 
+                                                method="to_markdown", load_by_default=True),
+                                DisplaySetting(selector="plot_data", type_="plot_data", 
+                                               method="plot_data", serialize_data=True)]
+        list_display_settings.extend(cls._decorators_settings())
+        return list_display_settings
+    
+    @classmethod
+    def _decorators_settings(cls) -> List[DisplaySetting]:
+        """ Return a list of objects describing how to call plot data displays. """
+        class_functions = inspect.getmembers(cls, inspect.isfunction)
+        method_names = []
+        for function_name, function in class_functions:
+            selector = 'plotdata'
+            serialize_data= ''
```

Comment:
serialize_data and load_by_default should be booleans and not strings.

But you can remove these two lines if you consider my next comment